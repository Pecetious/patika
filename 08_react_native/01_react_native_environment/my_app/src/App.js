import React from 'react';
import {SafeAreaView, StyleSheet, FlatList, Text} from 'react-native';
import news_data from './news_data.json';
import NewsCard from './components/NewsCard/NewsCard';
import NewsBanner from './components/NewsBanner/NewsBanner';
function App() {
  const handleRender = ({item}) => <NewsCard article={item} />;
  return (
    <SafeAreaView style={styles.container}>
      <Text style={styles.header_text}>NEWS</Text>
      <FlatList
        ListHeaderComponent={(<NewsBanner />)}
        keyExtractor={item => item.u_id.toString()}
        data={news_data}
        renderItem={handleRender}
      />
    </SafeAreaView>
  );
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#eceff1',
  },
  header_text: {
    fontWeight: 'bold',
    fontSize: 50,
  },
});
export default App;
