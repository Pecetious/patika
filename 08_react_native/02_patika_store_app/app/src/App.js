import React from 'react';
import {
  FlatList,
  SafeAreaView,
  StyleSheet,
  Text,
  TextInput,
} from 'react-native';
import store_data from './store_data.json';
import StoreCard from './components/StoreCard/StoreCard';
function App() {
  const handleRender = ({item}) => <StoreCard product={item} />;
  return (
    <SafeAreaView style={styles.container}>
      <Text style={styles.header_title}>patikastore</Text>
      <TextInput placeholder="Ara..." style={styles.search} />
      <FlatList data={store_data} renderItem={handleRender} numColumns={2} />
    </SafeAreaView>
  );
}
//tfjfhjd
const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#f5f5f5',
  },
  header_title: {
    color: 'purple',
    fontSize: 50,
    textTransform: 'uppercase',
    letterSpacing: -1,
    fontWeight: 'bold',
    paddingLeft: 20,
  },
  search: {
    fontSize: 22,
    borderRadius: 10,
    backgroundColor: 'lightgray',
    margin: 5,
    padding: 15,
  },
});
export default App;
