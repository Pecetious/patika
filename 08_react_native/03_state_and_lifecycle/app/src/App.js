import React, {useState} from 'react';
import {Button, FlatList, SafeAreaView, Switch, Text} from 'react-native';
const data = [
  {
    id: 0,
    name: 'cafe.exe',
    isFavorite: true,
  },
  {
    id: 1,
    name: 'KafeKafe',
    isFavorite: false,
  },
  {
    id: 2,
    name: 'BugG',
    isFavorite: false,
  },
  {
    id: 3,
    name: "Rock'n'Code",
    isFavorite: true,
  },
  {
    id: 4,
    name: 'do(drink)',
    isFavorite: false,
  },
  {
    id: 5,
    name: 'esc',
    isFavorite: false,
  },
];
function App() {
  const [cafeList, setCafeList] = useState(data);
  const [showFavorites, setShowFavorites] = useState(false);
  const handleFavorites = val => {
    setShowFavorites(val);
    val
      ? setCafeList(cafeList.filter(item => item.isFavorite))
      : setCafeList(data);
  };
  return (
    <SafeAreaView>
      <Switch
        value={showFavorites}
        onValueChange={change => handleFavorites(change)}
      />
      <FlatList
        data={cafeList}
        renderItem={({item}) => <Text>{item.name}</Text>}
      />
    </SafeAreaView>
  );
}

export default App;
