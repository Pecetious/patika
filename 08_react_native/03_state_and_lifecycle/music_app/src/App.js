import React, {useState} from 'react';
import {FlatList, SafeAreaView, StyleSheet, View} from 'react-native';
import music_data from './music-data.json';
import SongCard from './components/SongCard';
import SearchBar from './components/SearchBar';
function App() {
  const [filteredSongs, setFilteredSongs] = useState(music_data);
  const renderSong = ({item}) => <SongCard song={item} />;
  const renderSeparetor = () => <View style={styles.seperator} />;
  const handleSearch = text => {
    const filteredList = music_data.filter(song => {
      const searchedText = text.toLowerCase();
      const currentTitle = song.title.toLowerCase();
      return currentTitle.indexOf(searchedText) > -1;
    });
    setFilteredSongs(filteredList);
  };
  return (
    <SafeAreaView style={styles.container}>
      <SearchBar search={handleSearch} />
      <FlatList
        data={filteredSongs}
        renderItem={renderSong}
        ItemSeparatorComponent={renderSeparetor}
      />
    </SafeAreaView>
  );
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  seperator: {
    borderWidth: 1,
    borderColor: '#e0e0e0',
  },
});
export default App;
