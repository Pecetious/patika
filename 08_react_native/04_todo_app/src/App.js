import React, {useState} from 'react';
import {FlatList, SafeAreaView, StyleSheet, Text, View} from 'react-native';
import todos_data from './todos-data.json';
import TodoCard from './components/TodoCard';
import TodoInput from './components/TodoInput';
import {nanoid} from 'nanoid/non-secure';
function App() {
  const [todos, setTodos] = useState(todos_data);
  const renderTodos = ({item}) => (
    <TodoCard
      todo={item}
      toggle={handleToggleTodo}
      deleteTodo={handleDeleteTodo}
    />
  );
  const handleToggleTodo = id => {
    const updatedTodos = todos.map(todo =>
      todo.id === id ? {...todo, isDone: !todo.isDone} : todo,
    );
    setTodos(updatedTodos);
  };
  const handleDeleteTodo = id => {
    const updatedTodos = todos.filter(todo => todo.id !== id);
    setTodos(updatedTodos);
  };
  const handleAddTodo = todoName => {
    const todo = {
      id: nanoid(),
      name: todoName,
      isDone: false,
    };
    setTodos([...todos, todo]);
  };
  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.header}>
        <Text style={styles.header_text}>Todos</Text>
        <Text style={styles.header_text}>{todos.length}</Text>
      </View>
      <FlatList data={todos} renderItem={renderTodos} />
      <TodoInput click={handleAddTodo} />
    </SafeAreaView>
  );
}

export default App;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#262a30',
  },
  header: {
    flexDirection: 'row',
    margin: 10,
    padding: 10,
    justifyContent: 'space-between',
  },
  header_text: {
    color: "#f0950d",
    fontSize: 35,
    textTransform: 'uppercase',
    
  },
});
