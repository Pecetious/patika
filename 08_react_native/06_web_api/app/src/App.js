import React, {useState, useEffect} from 'react';
import {ActivityIndicator, FlatList, SafeAreaView, Text} from 'react-native';
import axios from 'axios';
import UserCard from './components/UserCard';
function App() {
  const [userList, setUserList] = useState([]);
  const [isLoading, setIsLoading] = useState(true);
  const handleFetch = async () => {
    const {data} = await axios.get(
      'https://jsonplaceholder.typicode.com/users',
    );
    setUserList(data);
    setIsLoading(false);
  };
  useEffect(() => {
    handleFetch();
  }, []);
  const renderUser = ({item}) => <UserCard user={item} />;
  return (
    <SafeAreaView>
      <Text>Hello World</Text>
      {!isLoading ? (
        <FlatList data={userList} renderItem={renderUser} numColumns="2" />
      ) : (
        <ActivityIndicator size="large" />
      )}
    </SafeAreaView>
  );
}

export default App;
