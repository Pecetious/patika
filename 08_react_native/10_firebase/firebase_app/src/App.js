import React from 'react';
import {Button, SafeAreaView, Text} from 'react-native';
import auth from '@react-native-firebase/auth';
import {firebase} from '@react-native-firebase/database';
function App() {
  // const signUp = () => {
  //   auth()
  //     .createUserWithEmailAndPassword(
  //       'jane.doe@example.com',
  //       'SuperSecretPassword!',
  //     )
  //     .then(() => {
  //       console.log('User account created & signed in!');
  //     })
  //     .catch(error => {
  //       if (error.code === 'auth/email-already-in-use') {
  //         console.log('That email address is already in use!');
  //       }

  //       if (error.code === 'auth/invalid-email') {
  //         console.log('That email address is invalid!');
  //       }

  //       console.error(error);
  //     });
  // };
  const checkDb = () => {
    const reference = firebase
      .app()
      .database(
        'https://reactnative-b04df-default-rtdb.europe-west1.firebasedatabase.app/',
      )
      .ref('cars/');
    reference
      .once('value')
      .then(snapshot => console.log(snapshot.val()))
      .catch(err => console.log(err));
  };
  const setDB = () => {
    const reference = firebase
      .app()
      .database(
        'https://reactnative-b04df-default-rtdb.europe-west1.firebasedatabase.app/',
      )
      .ref('cars/rentable');
    reference.set({
      brand: 'Audi',
      model: 'R8',
      price: 128,
    });
  };
  const pushDB = () => {
    const reference = firebase
      .app()
      .database(
        'https://reactnative-b04df-default-rtdb.europe-west1.firebasedatabase.app/',
      )
      .ref('cars/rentable');
    reference.push({
      brand: 'Lamborghini',
      model: 'Huracan SVJ',
      price: 350,
    });
  };
  return (
    <SafeAreaView>
      <Text>HELLO FIREBASE</Text>
      <Button title="check db" onPress={checkDb} />
      <Button title="set db" onPress={setDB} />
      <Button title="push db" onPress={pushDB} />
    </SafeAreaView>
  );
}

export default App;
