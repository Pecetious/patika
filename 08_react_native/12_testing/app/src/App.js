import React, {useState} from 'react';
import {
  FlatList,
  SafeAreaView,
  StyleSheet,
  Text,
  TextInput,
} from 'react-native';
import Button from './components/Button';

function App() {
  const renderElements = ({item}) => <Text>{item}</Text>;
  const [text, setText] = useState('');
  const [list, setList] = useState([]);
  const addToList = () => {
    if (text === '') return;
    setList([...list, text]);
    setText('');
  };
  return (
    <SafeAreaView style={styles.container}>
      <Text>Testing</Text>
      <FlatList testID="item-list" data={list} renderItem={renderElements} />
      <TextInput
        testID="item-input"
        placeholder="add..."
        style={styles.input}
        value={text}
        onChangeText={setText}
      />

      <Button testID="item-button" title="Radyoaktif Boi" onClick={addToList} />
    </SafeAreaView>
  );
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  input: {
    backgroundColor: 'gray',
    margin: 10,
    padding: 10,
    borderRadius: 10,
    color: 'white',
  },
});
export default App;
