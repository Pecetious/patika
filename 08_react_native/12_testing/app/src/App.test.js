import {render, fireEvent} from '@testing-library/react-native';
import React from 'react';
import App from './App';

test('should match with snapshot', () => {
  const comp = render(<App />);
  expect(comp).toMatchSnapshot();
});

test('should add item to list', () => {
  const testText = 'This is a test item';
  const comp = render(<App />);

  const itemInput = comp.getByTestId('item-input');
  fireEvent.changeText(itemInput, testText);

  const itemButton = comp.getByTestId('button-touchable');
  fireEvent.press(itemButton);

  const itemList = comp.getByTestId('item-list').props.data;
  expect(itemList.length).toBe(1);
});

test('should add multiple items', () => {
  const testItems = [
    'Item 1',
    'Item 2',
    'Item 3',
    'Item 4',
    'Item 5',
    'Item 6',
  ];
  const comp = render(<App />);
  testItems.forEach(testText => {
    const itemInput = comp.getByTestId('item-input');
    fireEvent.changeText(itemInput, testText);
    const itemButton = comp.getByTestId('button-touchable');
    fireEvent.press(itemButton);
  });
  const itemList = comp.getByTestId('item-list').props.data;
  expect(itemList.length).toBe(testItems.length);
});

test('should not add item if input area is empty', () => {
  const comp = render(<App />);
  const itemButton = comp.getByTestId('button-touchable');
  fireEvent.press(itemButton);

  const itemList = comp.getByTestId('item-list').props.data;
  expect(itemList.length).toBe(0);
});
