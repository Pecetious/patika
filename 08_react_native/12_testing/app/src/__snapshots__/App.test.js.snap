// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
    }
  }
>
  <Text>
    Testing
  </Text>
  <RCTScrollView
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    testID="item-list"
    viewabilityConfigCallbackPairs={[]}
  >
    <View />
  </RCTScrollView>
  <TextInput
    onChangeText={[Function]}
    placeholder="add..."
    style={
      {
        "backgroundColor": "gray",
        "borderRadius": 10,
        "color": "white",
        "margin": 10,
        "padding": 10,
      }
    }
    testID="item-input"
    value=""
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "blue",
        "borderRadius": 10,
        "justifyContent": "center",
        "margin": 10,
        "opacity": 1,
        "padding": 10,
      }
    }
    testID="button-touchable"
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
      testID="button-title"
    >
      Radyoaktif Boi
    </Text>
  </View>
</RCTSafeAreaView>
`;
